test/test_unary_ufuncs.py

```diff
@@ -479,7 +479,7 @@ def _test_out_arg(self, op, input, output, expected, **kwargs):
             self.assertTrue(res is output)
             self.assertEqual(output, expected.to(output.dtype))
 
-    @ops(unary_ufuncs, dtypes=OpDTypes.supported)
+    @ops(list(filter(lambda op: op.supports_out, unary_ufuncs)), dtypes=OpDTypes.supported)
```

Comment:
@Kiyosora is this change intended?

Reply:
https://github.com/pytorch/pytorch/blob/b5647dd52b48a51fec1387382deb5b59c7651512/aten/src/ATen/native/native_functions.yaml#L280-L284

`real` and `imag` don't have out variants. However the `UnaryUfuncInfo` assumes that every unary operator has `out` (which is true except for these two). 
So for this `test_out_arg_all_dtypes` which tests `out` variant, we have to filter these two operators.

**Note**: `list` is important. `filter` returns generator which is emptied with the tests for first device_type. cc: @mruberry

Reply:
Raising a skip if the operation does not support out in the test with a message explaining the skip is probably simpler than implementing the filter in the decorator.

Reply:
Right. That makes sense!